sds/src/simple_event.rs

```diff
@@ -2,7 +2,10 @@ use std::collections::BTreeMap;
 
 use crate::{encoding::Utf8Encoding, Event, EventVisitor, Path, PathSegment};
 
+static FIRST_VISIT_KEY: &str = "message";
+
 /// A simple implementation of `Event`. This is meant for testing / demonstration purposes.
+/// It always start visiting elements with key == "message" first, then the rest of the keys in the map
```

Comment:
multi-pass v0 rely on ordering as well for now and expect the path to be scanned last. This will also break the implicit expectation related to filtered encoded fields [here](https://github.com/DataDog/sds-shared-library/blob/main/sds-java/src/main/java/dd/sds/Encoder.java#L52-L69).

Reply:
This constraint might disappear in the future as cleanup of the java migration (multi-pass won't rely anymore on ordering) but it's a hard blocker for now

Reply:
More generally, we should not have any assumption in the sds-engine on the input structure. In fact, the `message` field  in logs can be named differently in spans/rum events. In my opinion, this constraint, if we really want it, should  to defined a scanner option configured by users, not as a hard-coded argument.

Reply:
resolving this comment as I realised that this is only used for tests and not anywhere else